sql/wsrep_sst.cc

```diff
@@ -190,6 +190,7 @@ static void wsrep_sst_complete (THD*                thd,
 {
   Wsrep_client_service client_service(thd, thd->wsrep_cs());
   Wsrep_server_state::instance().sst_received(client_service, rcode);
+  wsrep_init_server_gtid();
```

Comment:
This call to `wsrep_init_server_gtid()` is too late. `sst_received()` will allow applier threads to proceed, so some transactions may be applied and committed before the GTID state is initialized. This call should be moved somewhere in the main thread where it is guaranteed that the initialization happens after SE init but before appliers are allowed to continue.